### DISCUSSION TITLE
How does the cookie table work?

### DISCUSSION THREAD

**Discussion by LushawnDev** [orestbida/cookieconsent#510] [General]
Forgive me if it's explained elsewhere, I have really tried looking!

I believe I understand that certain scripts are blocked/allowed based on the user settings by the `value` key in the block config if they correspond with the data attribute on a script tag.

But is the `cookie_table` automatically generated? I can see there is an option for `is_regex` but I'm not entirely sure what this does. 

In the starter config there are 2 cookies set out in the `cookie_table`. They have "description..." and expiry dates set - should I expect these to be automatically populated or do I need to enter everything in the `cookie_table` myself?

Thanks for any help and accepting my ignorance! And thank you to the creators for creating a really sleek and lightweight open-source option for cookie banners - it's so much nicer than the mainstream options available!

**Comment by orestbida** [comment]
Hi @LushawnDev,

you have to manually specify all cookies inside `cookie_table`. 

> I can see there is an option for is_regex but I'm not entirely sure what this does.

The `is_regex` option is handy if you want to match multiple cookies without specifying them individually. 

This matches all cookies starting with `'_ga'`
```javascript
{
    col1: '^_ga',
    col2: 'yourdomain.com',
    col3: 'description ...',
    is_regex: true
}
```
This only matches the specific `'_ga'` cookie:
```javascript
{
    col1: '_ga',
    col2: 'yourdomain.com',
    col3: 'description ...'
}
```

You can customize the table (add/remove columns). The only requirement for the autoclear function to work is that the first column must contain the name of the cookie.

**Reply by asecondwill** [reply]
worth noting this in the docs i think.  I had the same confusion thinking it would add cookies from the script tags marked with  data-cookiecategory="analytics" to the analytics table.  Silly of me perhaps.  how would it know.

**Reply by orestbida** [reply]
Added this to the [Readme.md#how-to-clear-cookies](https://github.com/orestbida/cookieconsent/tree/v2.9#how-to-clear-cookies).
